src/pointfree/head.spec.js

```diff
@@ -72,3 +72,19 @@ test('head pointfree List', t => {
 
   t.end()
 })
+
+test('head pointfree iterable', t => {
+  const empty = testIterable(0, 0, 1)
+  const one = testIterable(1, 1, 1)
+  const two = testIterable(1, 2, 1)
+
+  t.equals(head(empty).type(), M.type(), 'returns a Maybe on empty')
+  t.equals(head(one).type(), M.type(), 'returns a Maybe on one element array')
+  t.equals(head(two).type(), M.type(), 'returns a Maybe on two element array')
+
+  t.equals(head(empty).option('Nothing'), 'Nothing', 'returns a Nothing on empty')
+  t.equals(head(one).option('Nothing'), 1, 'returns `1` on single element iterable')
+  t.equals(head(two).option('Nothing'), 1, 'returns `1` on dual element iterable')
```

Comment:
IMHO opinion it should be better to test a different value than 1 for the iterator with two elementas. The same idea we test on array and string.

Reply:
updated